drivers/pppos.c

```diff
@@ -448,46 +477,21 @@ static void pppos_mainLoop(void* _state)
 			log_info("open success!");
 		}
 
-		serial_set_non_blocking(state->fd);
-		if (at_disconnect(state->fd) != AT_RESULT_OK)
+		if (at_setup(state)) {
+			serial_close(state->fd);
 			goto fail;
-
-		mutexLock(state->lock);
-		while (!state->apn[0])
-			condWait(state->cond, state->lock, 0);
-		mutexUnlock(state->lock);
-
-		const char** at_cmd = at_init_cmds;
-		while (*at_cmd) {
-			if ((res = at_send_cmd(state->fd, *at_cmd, 3000)) != AT_RESULT_OK) {
-				log_warn("failed to initialize modem (cmd=%s), res=%d, retrying", *at_cmd, res);
-				goto fail;
-			}
-
-			at_cmd += 1;
 		}
 
-		{ /* Configure APN */
-			char at_set_apn[256];
-			if (snprintf(at_set_apn, sizeof(at_set_apn), "AT+CGDCONT=1,\"IP\",\"%s\"\r\n", state->apn) >= sizeof(at_set_apn)) {
-				log_error("APN name too long");
-				goto fail;
-			}
-
-			if ((res = at_send_cmd(state->fd, at_set_apn, 3000)) != AT_RESULT_OK) {
-				log_warn("failed to set APN, retrying");
-				goto fail;
-			}
+		if ((res = at_send_cmd(state->fd, "AT+CGATT=0\r\n", 3000)) != AT_RESULT_OK) {
```

Comment:
The modem utility has already been resetting the modem (via power up/enable pins) on every connection. I have added `ATZ` to `at_init_cmds` anyway in case there is a change that will make the serial device not reset the modem in hardware.